ush/python/pygfs/task/atmens_analysis.py

```diff
@@ -234,8 +232,8 @@ def get_bkg_dict(self, task_config: Dict[str, Any]) -> Dict[str, List[str]]:
             # accumulate directory list for member restart files
             dirlist.append(os.path.join(task_config.DATA, 'bkg', memchar))
 
-            # get FV3 RESTART files, this will be a lot simpler when using history files
-            rst_dir = os.path.join(task_config.comin_ges_atmens, memchar, 'atmos/RESTART')
+            # get FV3 restart files, this will be a lot simpler when using history files
+            rst_dir = os.path.join(task_config.COM_TOP_ENS_GES, memchar, 'model_data/atmos/restart')
```

Comment:
Instead of doing this, which basically reconstructs the atmos restart template, let's use `pygw.Template` and build paths here. Use this as a starting point, and then modify the config dictionaries that contain 'ROTDIR', etc. if I've gotten any wrong.

At the top of the script, you'll need to add the following import:
```
from pygw.template import Template, TemplateConstants
```

Then, change this part:
```suggestion
        template = task_config.COM_ATMOS_RESTART_TMPL
        tmpl_dict = {
            'ROTDIR': task_config.ROTDIR,
            'RUN': self.runtime_config.RUN,
            'YMD': self.runtime_config.PDY,
            'HH': self.runtime_config.cyc,
            }
        for imem in range(1, task_config.NMEM_ENKF + 1):
            memchar = f"mem{imem:03d}"
            # accumulate directory list for member restart files
            dirlist.append(os.path.join(task_config.DATA, 'bkg', memchar))

            # get FV3 restart files, this will be a lot simpler when using history files
            tmpl_dict['MEMDIR'] = memchar
            rst_dir = Template.substitute_structure(template, TemplateConstants.DOLLAR_CURLY_BRACE, tmpl_dict.get)
            os.makedirs(rst_dir, mode=0o775, exist_ok=True)
```
This will take the template and use the `tmpl_dict` dictionary to replace variables (defined as being within `${ ... }`) with entries from the dictionary.

Reply:
Oops, this was supposed to replace back to the start of the loop.

Reply:
Line 250 aborts with the traceback

```
Traceback (most recent call last):
  File "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/newcom/scripts/exglobal_atmens_analysis_initialize.py", line 25, in <module>
    AtmEnsAnl.initialize()
  File "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/newcom/ush/python/pygw/src/pygw/logger.py", line 264, in wrapper
    retval = func(*args, **kwargs)
  File "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/newcom/ush/python/pygfs/task/atmens_analysis.py", line 94, in initialize
    FileHandler(self.get_bkg_dict(AttrDict(self.task_config))).sync()
  File "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/newcom/ush/python/pygw/src/pygw/logger.py", line 264, in wrapper
    retval = func(*args, **kwargs)
  File "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/newcom/ush/python/pygfs/task/atmens_analysis.py", line 257, in get_bkg_dict
    rst_dir = Template.substitute_structure(template, TemplateConstants.DOLLAR_CURLY_BRACE, tmpl_dict.get)
  File "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/newcom/ush/python/pygw/src/pygw/template.py", line 115, in substitute_structure
    get_value)
  File "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/newcom/ush/python/pygw/src/pygw/template.py", line 87, in substitute_string
    variable, v)
TypeError: an integer is required (got type str)
```
This occurs while processing `mem001`.  The contents of `tmpl_dict` are 

`{'ROTDIR': '/scratch2/NCEPDEV/stmp1/Russ.Treadon/comrot/prufscom', 'RUN': 'enkfgdas', 'YMD': datetime.datetime(2021, 12, 21, 0, 0), 'HH': 0, 'MEMDIR': 'mem001'}
`